exporters/include/opentelemetry/exporters/common/http/http_client_curl.h

```diff
@@ -0,0 +1,227 @@
+#pragma once
+
+#include "http_operation_curl.h"
+
+#include <map>
+#include <string>
+#include <vector>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporters
+{
+namespace common
+{
+namespace http
+{
+namespace curl
+{
+
+namespace http_sdk                 = opentelemetry::sdk::common::http;
+const http_sdk::StatusCode Http_Ok = 200;
+
+class Request : public http_sdk::Request
+{
+public:
+  Request() : method_(http_sdk::Method::Get), uri_("/") {}
+
+  void SetMethod(http_sdk::Method method) noexcept override { method_ = method; }
+
+  void SetBody(http_sdk::Body &body) noexcept override { body_ = std::move(body); }
+
+  void AddHeader(nostd::string_view name, nostd::string_view value) noexcept override
+  {
+    headers_.insert(std::pair<nostd::string_view, nostd::string_view>(name, value));
+  }
+
+  void ReplaceHeader(nostd::string_view name, nostd::string_view value) noexcept override
+  {
+    // erase matching headers
+    auto range = headers_.equal_range(name);
+    headers_.erase(range.first, range.second);
+    AddHeader(name, value);
+  }
+
+  virtual void SetUri(nostd::string_view uri) noexcept override { uri_ = uri; }
```

Comment:
nit: `virtual` key word is not necessary here because there is `override`?

Reply:
It was on purpose - to allow possibility of extending the the `http curl` implementation ( either by sdk developers, or as part of gtest ).